src/components/signup.jsx

```diff
@@ -1,15 +1,29 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
+import axios from 'axios';
 
 class Signup extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
       username: '',
-      password: ''
+      password: '',
+      signupFailed: 'hidden',
     };
   }
 
+  trySignup(signUpObject) {
```

Comment:
Add to backlog: let's find a way to refactor to combine some of the signup and login logic, since they're so similar.